document `-enable-plugin-cache` flag (#34644)
hashicorp/terraform: website/docs/cli/commands/providers/lock.mdx

  There is more detail on this option in the following section.

> * `-enable-plugin-cache` -  Enable the usage of the [globally configured plugin cache](/terraform/cli/config/config-file#provider-plugin-cache).

Reviewer: would be good to just add a clarification as to why it doesn't do this automatically (or in the same way the `init` command does).

  There is more detail on this option in the following section.

* `-enable-plugin-cache` -  Enable the usage of the [globally configured plugin cache](/terraform/cli/config/config-file#provider-plugin-cache).

>   This will speed up the locking process. This is not enabled by default since
>   the providers won't be loaded from an authoritative source. As the
>   `terraform provider lock` command is used to ensure no untrusted provider
>   versions can be used, e.g. by adding them to the plugin cache or a mirror
>   installing the plugins from cache is potentially risky.

Reviewer: ```suggestion
  This will speed up the locking process. This is not enabled by default since
  the the plugin cache is not an authoritative source. As the
  `terraform provider lock` command is used to ensure no untrusted provider
  versions can be used installing the plugins from the cache is considered risky.
```

I've tried to make it a bit more verbose, and avoided using words like "potentially" without explaining the consequences behind it.
